src/codec.rs

```diff
@@ -147,6 +147,33 @@ impl Codec for u64 {
     }
 }
 
+impl Codec for String {
+    fn encode(&self, buffer: &mut Vec<u8>) -> Result<(), CodecError> {
+        let string_bytes = self.as_bytes();
+        (string_bytes.len() as u16).encode(buffer)?;
```

Comment:
What happens here if the string is too long? Should we catch that explicitly with an encoding error?

Reply:
yeah we should 👍